webapp/masterclasses.py

```diff
@@ -48,23 +81,45 @@ def get_upcoming_sessions():
         .order_by(nullslast(UpcomingSession.date.desc()))
         .all()
     )
-    for i in range(len(session)):
-        if session[i].date:
-            session[i].date = format_date(session[i].date)
+    for s in session:
+        if s.date:
+            s.date = format_date(s.date)
     return session
 
 
 def get_previous_sessions():
     session = (
         db_session.query(PreviousSession).order_by(PreviousSession.date.desc()).all()
     )
-    for i in range(len(session)):
-        session[i].date = format_date(session[i].date)
+    for s in session:
+        s.date = format_date(s.date)
+        s.video_id = get_video_id(s.recording)
```

Comment:
We seem to do this multiple times, maybe have an enrich session function that formats the date and sets the video ID in case a recording property exists instead of maintaining 3 copies of this code bit?